src/Components/Components/src/ComponentsActivitySource.cs

```diff
@@ -0,0 +1,177 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+
+namespace Microsoft.AspNetCore.Components;
+
+/// <summary>
+/// This is instance scoped per renderer
+/// </summary>
+internal class ComponentsActivitySource
+{
+    internal const string Name = "Microsoft.AspNetCore.Components";
+    internal const string OnCircuitName = $"{Name}.CircuitStart";
+    internal const string OnRouteName = $"{Name}.RouteChange";
+    internal const string OnEventName = $"{Name}.HandleEvent";
+
+    private ActivityContext _httpContext;
+    private ActivityContext _circuitContext;
+    private string? _circuitId;
+    private ActivityContext _routeContext;
+
+    private ActivitySource ActivitySource { get; } = new ActivitySource(Name);
+
+    public static ActivityContext CaptureHttpContext()
+    {
+        var parentActivity = Activity.Current;
+        if (parentActivity is not null && parentActivity.OperationName == "Microsoft.AspNetCore.Hosting.HttpRequestIn" && parentActivity.Recorded)
+        {
+            return parentActivity.Context;
+        }
+        return default;
+    }
```

Comment:
Layering wise this shouldn't go here.

`Microsoft.AspNetCore.Components` shoudn't have a dependency on `Microsoft.AspNetCore.Hosting`

Reply:
Please be more specific, what do you want to see instead ?

Reply:
Are you trying to get the activity of the current request here? There is a feature in the features collection for doing that: https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.features.ihttpactivityfeature.activity?view=aspnetcore-9.0

Reply:
Can I get that from DI without passing HttpContext thru all SignalR layers ? Or from some TLS ?

Reply:
It comes from HttpContext.